src/components/NewTagDialog.tsx

```diff
@@ -0,0 +1,473 @@
+              baseCommitId={baseCommitId}
+              trans={props.trans}
+              setRef={node => {
+                nodes.current[commit.commit] = node;
+              }}
+              updateBaseCommitId={updateBaseCommitId}
+            ></SingleCommitNode>
+          );
+        })}
+      </ol>
+    </div>
+  );
+};
+
+export const SingleCommitNode: React.FunctionComponent<ISingleCommitProps> = (
+  props: ISingleCommitProps
+): React.ReactElement => {
+  const { baseCommitId, updateBaseCommitId } = props;
+  const isLatest = props.commit.commit === baseCommitId;
+
+  let isBold;
+  if (isLatest) {
+    isBold = true;
+  }
+  /**
+   * Returns a callback which is invoked upon clicking a commit name.
+   *
+   * @param commit - commit
+   * @returns callback
+   */
+  const onCommitClickFactory = (commitId: string) => {
+    return onClick;
+
+    /**
+     * Callback invoked upon clicking a commit.
+     *
+     * @private
+     * @param event - event object
+     */
+    function onClick(): void {
+      updateBaseCommitId(commitId);
+    }
+  };
+
+  return (
+    <li
+      id={props.commit.commit}
+      ref={props.setRef}
+      className={classes(
+        commitWrapperClass,
+        isLatest ? activeListItemClass : null
+      )}
+      title={props.trans.__(
+        'Create a new tag pointing to commit %1: %2 by %3',
+        props.commit.commit in Git.Diff.SpecialRef
+          ? Git.Diff.SpecialRef[+props.commit.commit]
+          : props.commit.commit.slice(0, 7),
+        props.commit.commit_msg,
+        props.commit.author
+      )}
+      onClick={onCommitClickFactory(props.commit.commit)}
+    >
+      <div className={commitHeaderClass}>
+        <span
+          className={classes(
+            commitHeaderItemClass,
+            isBold ? commitItemBoldClass : null
+          )}
+        >
+          {props.commit.author}
+        </span>
+        <span
+          className={classes(
+            commitHeaderItemClass,
+            isBold ? commitItemBoldClass : null
+          )}
+        >
+          {+props.commit.commit in Git.Diff.SpecialRef
+            ? Git.Diff.SpecialRef[+props.commit.commit]
+            : props.commit.commit.slice(0, 7)}
+        </span>
+        <span
+          className={classes(
+            commitHeaderItemClass,
+            isBold ? commitItemBoldClass : null
+          )}
+        >
+          {props.commit.date}
+        </span>
+      </div>
+      <div
+        className={classes(
+          commitBodyClass,
+          isBold ? commitHeaderBoldClass : null
+        )}
+      >
+        {props.commit.commit_msg}
+      </div>
+    </li>
+  );
+};
+
+export const NewTagDialogBox: React.FunctionComponent<INewTagDialogProps> = (
+  props: INewTagDialogProps
+): React.ReactElement => {
+  const [nameState, setNameState] = React.useState('');
+  const [baseCommitIdState, setBaseCommitIdState] = React.useState(
+    props.pastCommits[0]?.commit || null
```

Comment:
It is better to use the non-nullish operator:

```suggestion
    props.pastCommits[0]?.commit ?? null
```

The reason is that if the left value is falsy (like 0 or empty string), the non-nullish operator will use that falsy value not the fallback in the opposite of the _or_ operator.